### PR TITLE
Fix Pod factory fixture

### PR DESCRIPTION
**Pull request by RazTamir**
Signed-off-by: ratamir <ratamir@redhat.com>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -357,13 +358,11 @@ def factory(
                 interface_type=interface,
             )
             assert pod_obj, "Failed to create PVC"
-            helpers.wait_for_resource_state(pod_obj, constants.STATUS_RUNNING)
-            pod_obj.reload()
+            instances.append(pod_obj)
```
Comment (petr-balogh) [inline tests/conftest.py]: If you will move indent 1 tab to the left here you can remove one from line 351 an will have the same behavior and 1 line less ;)

Comment (RazTamir) [reply tests/conftest.py]: done

**Comment by RazTamir** [comment]
Verification link:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/474/console

**Comment by vasukulkarni** [comment]
Issue please? ,  I can amend the commit message, but not sure what issue was fixed?

**Comment by RazTamir** [comment]
It fixes the problem when the pod creation succeeded but failed to reach running state and the pod is not added to instances list to be removed in teardown.
No issue was created

**Comment by vasukulkarni** [comment]
I have amended the commit message to include that.

**Pull request merged**